### PR TITLE
Support ligatures in libtxt

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@cbracken @chinmaygarde

File: third_party/txt/src/txt/paragraph.cc [hunk 1]
```diff
@@ -440,7 +437,25 @@ void Paragraph::Layout(double width, bool force) {
...
+          // are joined into this glyph.
+          size_t ligature_end = character_index + 1;
+          for (; ligature_end < text_count; ++ligature_end) {
+            if (layout.getCharAdvance(ligature_end) != 0)
+              break;
+          }
+          size_t ligature_width = ligature_end - character_index;
+          prev_char_advance = glyph_advance / ligature_width;
+          // Compute positions for the additional characters in the ligature.
+          for (size_t i = 1; i < ligature_width; i++) {
```
Comment (cbracken) [inline third_party/txt/src/txt/paragraph.cc]: `++i` for consistency... not that the compiler won't optimise that away anyway :-)

**Comment by cbracken** [comment]
lgtm

**Comment by jason-simmons** [comment]
Some fonts do include a table of caret positions within the ligature (although most of the fonts I tried on an Android device did not have this table).

HarfBuzz does have an API for retrieving the positions on fonts that support it (hb_ot_layout_get_ligature_carets).  Distributing the positions evenly within the ligature yields decent results for now, but we could try using this API if we encounter fonts where it helps.

**Pull request merged**